crates/derive/src/batch/span_batch/batch.rs

```diff
@@ -42,6 +42,11 @@ impl SpanBatch {
         self.batches[0].timestamp
     }
 
+    /// Returns the ending timestamp for the span batch.
+    pub fn end_timestamp(&self) -> u64 {
+        self.batches.last().map_or(0, |b| b.timestamp)
```

Comment:
I think we should follow the same behavior as `timestamp` and panic if `batches` is empty. Getting `0` could cause some weird UB.